public/app/features/query/components/QueryEditorRow.tsx

```diff
@@ -81,12 +81,11 @@ export class QueryEditorRow<TQuery extends DataQuery> extends PureComponent<Prop
     showingHelp: false,
   };
 
-  componentDidMount() {
+  async componentDidMount() {
     const { data, query } = this.props;
     const dataFilteredByRefId = filterPanelDataToQuery(data, query.refId);
     this.setState({ data: dataFilteredByRefId });
-
-    this.loadDatasource();
+    await this.loadDatasource();
```

Comment:
why is this change needed?

Reply:
This was leftover from when I was trying to avoid the state change error. I'll remove it.